### ISSUE TITLE
Questions about intended tool behavior.

### ISSUE THREAD

**Issue by Phylogeny** [Direwolf20-MC/BuildingGadgets#83]
Is it intended behavior that:
1. Blocks can be exchanged/placed in adventure mode.
2. Unbreakable blocks, like bedrock, can be exchanged/placed when in survival/adventure mode.
3. When an effect block is copied or is set as the selected block of an exchanger or builder, if it is exchanged/placed/pasted while in survival mode, a construction block (hardness 2) is exchanged/placed/pasted instead. But if the same is done while in creative mode, the actual effect block (hardness 20) is exchanged/placed/pasted.

**Comment by Direwolf20-MC** [comment]
1) probably not, you can't place blocks in adventure mode can you?
2) Yes
3) Yes

**Comment by Phylogeny** [comment]
Yeah, held item stacks can only place blocks on certain blocks specified by the canPlaceBlock on NBT tag. If the item stack lacks that tag, or if it has no data, then it cannot be used to place any block in the world.
